src/ansiblelint/rules/only_builtins.py

```diff
@@ -0,0 +1,69 @@
+"""Rule definition for usage of builtin actions only."""
+import sys
+from typing import Any, Dict, Optional, Union
+
+from ansiblelint.file_utils import Lintable
+from ansiblelint.rules import AnsibleLintRule
+
+# fqcn_builtins was added in 5.1.0 as FQCNBuiltinsRule, renamed to fqcn_builtins in 6.0.0
+from ansiblelint.rules.fqcn_builtins import builtins
+
+
+class OnlyBuiltinsRule(AnsibleLintRule):
+    """Use only builtin actions."""
+
+    id = "only-builtins"
+    severity = "MEDIUM"
+    description = "Check whether the playbook uses anything but ``ansible.builtin``"
+    tags = ["opt-in", "formatting", "experimental"]
+
+    def matchtask(
+        self, task: Dict[str, Any], file: Optional[Lintable] = None
+    ) -> Union[bool, str]:
+        fqcn_builtin = task["action"]["__ansible_module_original__"].startswith(
+            "ansible.builtin."
+        )
+        non_fqcn_builtin = task["action"]["__ansible_module_original__"] in builtins
+        return not fqcn_builtin and not non_fqcn_builtin
+
+
+# testing code to be loaded only with pytest or when executed the rule file
+if "pytest" in sys.modules:
+
+    import pytest
+
+    from ansiblelint.testing import RunFromText  # pylint: disable=ungrouped-imports
+
+    SUCCESS_PLAY = """
+- hosts: localhost
+  tasks:
+  - name: shell (fqcn)
+    ansible.builtin.shell: echo This rule should not get matched by the only-builtins rule
+    """
+
+    FAIL_PLAY = """
+- hosts: localhost
+  tasks:
+  - name: fake_namespace.fake_collection.fake_module
+    fake_namespace.fake_collection.fake_module:
+      name: just a fake
+    """
+
+    @pytest.mark.parametrize(
+        "rule_runner", (OnlyBuiltinsRule,), indirect=["rule_runner"]
+    )
+    def test_only_builtin_fail(rule_runner: RunFromText) -> None:
```

Comment:
That is not a proper test, also test for message but in order to not have
syntax error, try to use something like https://docs.ansible.com/ansible/latest/collections/community/general/ini_file_module.html -- which is not included in ansible-core but is likely already installed.

The idea is to to ensure that the rule can be triggered.

Reply:
As noted in https://github.com/ansible/ansible-lint/pull/2060/files/7e6b349128bb2b676ad937a691b256fa5e754a25#diff-8b028ab60fefe40d188162c38df91f412a790f6ceae38c194a61a66992c7e5a4, I couldn't find a module that would not end up being not found at all.

I've pushed a version with `ini_file` but I don't think this will make the tests any happier. CI runs in a clean environment, where no collections are installed.

Reply:
Okay, I've explicitly added `ansible.posix` to the `requirements.yml` and am now using that.

If you have a better suggestion for a collection to pull in -- or know one that would be present (`community.general` is not, see https://github.com/ansible/ansible-lint/runs/6211280681?check_suite_focus=true) -- I'll happily change the test to use that.